peakdet/editor.py

```diff
@@ -60,29 +60,36 @@ def __init__(self, data):
         delete = functools.partial(self.on_edit, method="delete")
         reject = functools.partial(self.on_edit, method="reject")
         insert = functools.partial(self.on_edit, method="insert")
+
+        # Check matplotlib version rectprops is deprecated with matplotlib 3.5.0 and then obsolete 
+        if matplotlib.__version__ >= '3.5.0':
```

Comment:
You need to explicitly `import matplotlib` for this. The current imports don't cover this. This is also indicated in the `pre-commit` fail